### ISSUE TITLE
add support for concurrent.futures

### ISSUE THREAD

**Issue by mmckerns** [uqfoundation/pathos#90]
the `pathos` interface is nearly a match to `concurrent.futures`, and it seems to be preferred over `multiprocessing` as the standard.

**Comment by PawelTroka** [comment]
@mmckerns Any updates on this?

**Comment by mmckerns** [comment]
No, I haven't had time for it.  In theory, it should be an easy PR -- however, I'm short on time at the moment.

**Comment by evanbiederstedt** [comment]
+1 

This would be very very useful

**Comment by colibri17** [comment]
+1 Agree

**Comment by HerculesJack** [comment]
+1

**Comment by matthewyangcs** [comment]
+1 please!

**Comment by ddelange** [comment]
fwiw, in case you're in the `asyncio` world and you're in need of `dill`, this gist might be useful to you (particularly the `AioPool` part):
https://gist.github.com/ddelange/643fbb791b398783c04d1ceb90102163

See also the PR I liked above in 2021. It's not a drop in replacement of ProcessPoolExecutor, but the gist mostly gets the job done for me.

**Comment by mmckerns** [comment]
I just had this come up in a funded project yesterday, so this may get prioritized...

**Comment by ddelange** [comment]
@mmckerns that's cool to hear! the unresolved [comment](https://github.com/dano/aioprocessing/pull/36#discussion_r631178933) might be relevant for the implementation

**Comment by rachtsingh** [comment]
One important difference between `multiprocessing.Pool` and `concurrent.futures.ProcessPoolExecutor` is that the [former hangs if a worker process is killed](https://github.com/python/cpython/issues/66587). Does the pathos/multiprocess project have a plan to handle that behavior (or is it already handled in `multiprocess.Pool`)?

As an alternative for those looking, it looks like the `joblib/loky` project does try to handle this? I'll give it a shot.

**Comment by ddelange** [comment]
I think that's https://github.com/uqfoundation/pathos/issues/265
